internal/internal_workflow_client.go

```diff
@@ -91,21 +91,38 @@ type (
 		// GetID return workflow ID, which will be same as StartWorkflowOptions.ID if provided.
 		GetID() string
 
-		// GetRunID return the first started workflow run ID (please see below) - empty string if no such run
+		// GetRunID return the first started workflow run ID (please see below) -
+		// empty string if no such run. Note, this value may change after Get is
+		// called if there was a later run for this run.
 		GetRunID() string
 
-		// Get will fill the workflow execution result to valuePtr,
-		// if workflow execution is a success, or return corresponding,
-		// error. This is a blocking API.
+		// Get will fill the workflow execution result to valuePtr, if workflow
+		// execution is a success, or return corresponding error. This is a blocking
+		// API.
+		//
+		// This call will follow execution runs to the latest result for this run
+		// instead of strictly returning this run's result. This means that if the
+		// workflow returned ContinueAsNewError, has a more recent cron execution,
+		// or has a new run ID on failure (i.e. a retry), this will wait and return
+		// the result for the latest run in the chain. To strictly get the result
+		// for this run without following to the latest, use GetWithOptions and set
+		// the DisableFollowingRuns option to true.
 		Get(ctx context.Context, valuePtr interface{}) error
 
-		// NOTE: if the started workflow return ContinueAsNewError during the workflow execution, the
-		// return result of GetRunID() will be the started workflow run ID, not the new run ID caused by ContinueAsNewError,
-		// however, Get(ctx context.Context, valuePtr interface{}) will return result from the run which did not return ContinueAsNewError.
-		// Say ExecuteWorkflow started a workflow, in its first run, has run ID "run ID 1", and returned ContinueAsNewError,
-		// the second run has run ID "run ID 2" and return some result other than ContinueAsNewError:
-		// GetRunID() will always return "run ID 1" and  Get(ctx context.Context, valuePtr interface{}) will return the result of second run.
-		// NOTE: DO NOT USE client.ExecuteWorkflow API INSIDE A WORKFLOW, USE workflow.ExecuteChildWorkflow instead
+		// GetWithOptions will fill the workflow execution result to valuePtr, if
+		// workflow execution is a success, or return corresponding error. This is a
+		// blocking API.
+		GetWithOptions(ctx context.Context, valuePtr interface{}, options WorkflowRunGetOptions) error
+	}
+
+	// WorkflowRunGetOptions are options for WorkflowRun.GetWithOptions.
+	WorkflowRunGetOptions struct {
+		// DisableFollowingRuns, if true, will not follow execution chains to the
+		// latest run. By default when this is false, getting the result of a
+		// workflow may not use the literal run ID but instead follow to later runs
+		// if the workflow returned a ContinueAsNewError, has a later cron, or is
+		// retried on failure.
+		DisableFollowingRuns bool
```

Comment:
Should I fail if this was set to true but a run ID was not provided? Is there a use case for not providing a run ID but still not following to the next run?

Reply:
We don't in TS. I can't think of a reason not to allow it, saves looking up the last run ID for a workflow ID.

Reply:
Can we change this setting name to `FollowRuns` and still default to `true`? It's easier to follow.

Reply:
> Can we change this setting name to FollowRuns and still default to true? It's easier to follow.

I am afraid not in Go. You can't default fields, their default is always the zero value which for booleans is false. This is why for all Go structs, you specifically build the field where false is the reasonable default.

Reply:
Yeah, I get it. I thought that we could check if the options struct is null and default to true but we want empty struct to also default to true.

Reply:
This options struct is never null inside the outer options (it's not a pointer). And even if it could be nullable, we want to support people creating it and just setting the fields they care about instead of trying to differentiate between unset and explicit false.

Reply:
Cool, I agree this is the best approach